azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/profile/VisualStudioCodeProfileRetriever.java

```diff
@@ -0,0 +1,118 @@
+                map.put(CLOUD, userSettings.get(AZURE_CLOUD).asText());
+            }
+            if (userSettings.has(AZURE_TENANT) && StringUtils.isNotBlank(userSettings.get(AZURE_TENANT).asText())) {
+                map.put(TENANT, userSettings.get(AZURE_TENANT).asText());
+            }
+        }
+        map.putIfAbsent(CLOUD, "AzureCloud");
+        return map;
+    }
+
+    private static VisualStudioCodeAccountProfile getVsCodeAccountProfile(JsonNode userSettings, String cloud) throws InvalidConfigurationException {
+        List<String> filteredSubscriptions = new ArrayList<>();
+
+        if (userSettings.has(AZURE_RESOURCE_FILTER)) {
+            for (JsonNode filter : userSettings.get(AZURE_RESOURCE_FILTER)) {
+                String[] tenantAndSubsId = StringUtils.split(filter.asText(), "/");
+                if (tenantAndSubsId.length == 2) {
+                    filteredSubscriptions.add(tenantAndSubsId[1]);
+                } else {
+                    throw new InvalidConfigurationException(
+                            String.format("Invalid 'azure.resourceFilter' settings '%s' in VSCode settings.json.", filter.asText()));
+                }
+            }
+        }
+        return new VisualStudioCodeAccountProfile(cloud, filteredSubscriptions.toArray(new String[0]));
+    }
+
+    /**
+     * Code copied from https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/VisualStudioCacheAccessor.java#L32
+     * , since it doesn't support profile, we need to get the setting for user
+     * selected subscription
+     */
+    private static JsonNode getUserSettings() {
+        JsonNode output;
+        String homeDir = System.getProperty("user.home");
+        String settingsPath = "";
+        ObjectMapper mapper = new ObjectMapper();
+        mapper.configure(JsonReadFeature.ALLOW_UNESCAPED_CONTROL_CHARS.mappedFeature(), true);
+        mapper.configure(JsonReadFeature.ALLOW_JAVA_COMMENTS.mappedFeature(), true);
+        try {
+            if (Platform.isWindows()) {
+                settingsPath = Paths.get(System.getenv("APPDATA"), "Code", "User", "settings.json")
+                        .toString();
+            } else if (Platform.isMac()) {
+                settingsPath = Paths.get(homeDir, "Library",
+                        "Application Support", "Code", "User", "settings.json").toString();
+            } else if (Platform.isLinux()) {
+                settingsPath = Paths.get(homeDir, ".config", "Code", "User", "settings.json")
+                        .toString();
+            } else {
+                throw new CredentialUnavailableException(PLATFORM_NOT_SUPPORTED_ERROR);
+            }
+            File settingsFile = new File(settingsPath);
+            output = mapper.readTree(settingsFile);
+        } catch (Exception e) {
```

Comment:
`PLATFORM_NOT_SUPPORTED_ERROR` should not be swallowed.

Reply:
this method is from azure identity, will be removed after azure identity fixes the bug.